raster/r.neighbors/main.c

```diff
@@ -258,13 +306,19 @@ int main(int argc, char *argv[])
 	G_fatal_error(_("Neighborhood size must be odd"));
     ncb.dist = ncb.nsize / 2;
 
-    if (parm.weight->answer && flag.circle->answer)
+    if (strcmp(parm.weighting_function->answer, "none") == 1 && flag.circle->answer)
```

Comment:
`strcmp(parm.weighting_function->answer, "none") == 1` would not work because `strcmp(parm.weighting_function->answer, "none")` must not be zero, thus `strcmp(parm.weighting_function->answer, "none")`, in full
`if (strcmp(parm.weighting_function->answer, "none") && flag.circle->answer)`
should work

Reply:
Thanks for the hint/help with that! The suggestion is now implemented.